prow/entrypoint/run_test.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package entrypoint
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	"k8s.io/test-infra/prow/pod-utils/wrapper"
+)
+
+func TestOptions_Run(t *testing.T) {
+	var testCases = []struct {
+		name           string
+		args           []string
+		expectedLog    string
+		expectedMarker string
+		expectedErr    bool
+	}{
+		{
+			name:           "successful command",
+			args:           []string{"sh", "-c", "exit 0"},
+			expectedLog:    "",
+			expectedMarker: "0",
+			expectedErr:    false,
+		},
+		{
+			name:           "successful command with output",
+			args:           []string{"echo", "test"},
+			expectedLog:    "test\n",
+			expectedMarker: "0",
+			expectedErr:    false,
+		},
+		{
+			name:           "unsuccessful command",
+			args:           []string{"sh", "-c", "exit 12"},
+			expectedLog:    "",
+			expectedMarker: "12",
+			expectedErr:    true,
+		},
+		{
+			name:           "unsuccessful command with output",
+			args:           []string{"sh", "-c", "echo test && exit 12"},
+			expectedLog:    "test\n",
+			expectedMarker: "12",
+			expectedErr:    true,
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			tmpDir, err := ioutil.TempDir("", testCase.name)
+			if err != nil {
+				t.Errorf("%s: error creating temp dir: %v", testCase.name, err)
+			}
+			defer func() {
+				if err := os.RemoveAll(tmpDir); err != nil {
+					t.Errorf("%s: error cleaning up temp dir: %v", testCase.name, err)
+				}
+			}()
+
+			options := Options{
+				Args: testCase.args,
+				Options: &wrapper.Options{
+					ProcessLog: path.Join(tmpDir, "process-log.txt"),
+					MarkerFile: path.Join(tmpDir, "marker-file.txt"),
+				},
+			}
+
+			if err := options.Run(); err == nil && testCase.expectedErr {
+				t.Errorf("%s: expected an error but got none", testCase.name)
+			} else if err != nil && !testCase.expectedErr {
+				t.Errorf("%s: expected no error but got one: %v", testCase.name, err)
+			}
+
+			compareFileContents(testCase.name, options.ProcessLog, testCase.expectedLog, t)
+			compareFileContents(testCase.name, options.MarkerFile, testCase.expectedMarker, t)
+		})
+	}
+}
+
+func compareFileContents(name, file, expected string, t *testing.T) {
+	data, err := ioutil.ReadFile(file)
+	if err != nil {
+		t.Fatalf("%s: could not read file: %v", name, err)
+	}
+	if string(data) != expected {
```

Comment:
diff?

Reply:
What do you mean?

Reply:
I'm hoping you can magically conjure up a difflib that makes it easy for me to see the two lines in this huge string that change... so I can start using it :smiling_imp:

Reply:
IIRC the ObjectReflectDiff will actually do that, too